### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by ntfshard**

### CODE REVIEW

**Comment by ntfshard** [comment]
make test

**Comment by ntfshard** [comment]
make test

File: algorithms/kernel/assocrules/assoc_rules_apriori_tree.i [hunk 1]
```diff
@@ -157,7 +157,7 @@ struct hash_tree
         if (!is_child(node_idx, hash_res))
         {
             /* Update children mask of the root node if needed */
-            root[node_idx].mask |= (1 << hash_res);
+            root[node_idx].mask |= (__int64)(1) << hash_res;
```
Comment (Vika-F) [inline algorithms/kernel/assocrules/assoc_rules_apriori_tree.i]: Too much braces. It's better to have just __int64(1). Anyway, I'm Ok with the change.

File: algorithms/kernel/optimization_solver/lbfgs/lbfgs_dense_default_impl.i [hunk 2]
```diff
@@ -97,7 +97,7 @@ services::Status LBFGSKernel<algorithmFPType, defaultDense, cpu>::compute(
 
     if (useWolfeConditions)
     {
-        gradientFunction->sumOfFunctionsParameter->resultsToCompute |= objective_function::value;
+        gradientFunction->sumOfFunctionsParameter->resultsToCompute |= (DAAL_UINT64)(objective_function::value);
```
Comment (Vika-F) [inline algorithms/kernel/optimization_solver/lbfgs/lbfgs_dense_default_impl.i]: Too much braces again.

File: algorithms/kernel/assocrules/assoc_rules_apriori_tree.i (hunk 1 above)
Comment (ntfshard) [reply algorithms/kernel/assocrules/assoc_rules_apriori_tree.i]: Semantically it's different construtions. We must put type inside () due it's not 1 word.

Comment (ntfshard) [reply algorithms/kernel/assocrules/assoc_rules_apriori_tree.i]: (Type)1  - c-style cast
Type(1)  construction of Type with passed value to ctor's parameter

File: include/data_management/data_source/file_data_source.h [hunk 3]
```diff
@@ -196,7 +196,7 @@ class FileDataSource : public CsvDataSource<FeatureManager, SummaryStatisticsTyp
             return services::throwIfPossible(services::ErrorOnFileOpen);
         }
 #else
-        _file = fopen((char *)(fileName.c_str()), "r");
+        _file = std::fopen((char *)(fileName.c_str()), "r");
```
Comment (Vika-F) [inline include/data_management/data_source/file_data_source.h]: If this worked previously, then we have "using namespace std" or "#include <stdio.h>" somewhere in the includes. But I can't find any of those. I am Ok with the change, but can you please explain how this worked previously?

File: samples/java/hadoop/sources/LinearRegressionNormEq.java [hunk 4]
```diff
@@ -51,8 +51,14 @@
...
 
-    public static void main(String[] args) throws Exception {
-        int res = ToolRunner.run(new Configuration(), new LinearRegressionNormEq(), args);
+    public static void main(String[] args) {
+        int res = -1;
+        try {
+            res = ToolRunner.run(new Configuration(), new LinearRegressionNormEq(), args);
+        } catch (Exception e) {
+            System.out.println("Error");
+            e.printStackTrace();
```
Comment (Vika-F) [inline samples/java/hadoop/sources/LinearRegressionNormEq.java]: Let's print more meaningful diagnostic. I think that stack trace is not enough.
            System.out.println("Exception caught: " + e);
            System.out.println("Exception message: " + e.getMessage());
            System.out.println("Exception cause: " + e.getCause());
            e.printStackTrace();

Same comment to the rest of the samples.

File: include/data_management/data_source/file_data_source.h (hunk 3 above)
Comment (ntfshard) [reply include/data_management/data_source/file_data_source.h]: First of all, according to [iso cpp core guidelines](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rs-using-directive) we mustn't write `using namespace std` inside headers.
The second one, fopen without namespace is available for compatibility with C. We have included cstdio in the begging of the file. So we should use [std::fopen](https://en.cppreference.com/w/cpp/io/c/fopen)

Comment (Vika-F) [reply include/data_management/data_source/file_data_source.h]: I do know that "using namespace std" is prohibited. But I didn't know that fopen exists in global namespace. Thanks for explanation!

File: algorithms/kernel/assocrules/assoc_rules_apriori_tree.i (hunk 1 above)
Comment (ntfshard) [reply algorithms/kernel/assocrules/assoc_rules_apriori_tree.i]: replaced by static_cast

**Comment by KalyanovD** [comment]
restart

File: include/data_management/data_source/file_data_source.h (hunk 3 above)
Comment (averbukh) [reply include/data_management/data_source/file_data_source.h]: OK to have 'std::' here. The possible reason why it worked before is inclusion of stdio.h in another file that included here indirectly.

**Comment by ntfshard** [comment]
restart

**Comment by ntfshard** [comment]
make test

**Pull request merged**